### ISSUE TITLE
Uncaught TypeError: Cannot read property '0' of undefined from wheeled function

### ISSUE THREAD

**Issue by martinek** [d3/d3-zoom#41]
I probably came upon some kind of edge case. When zooming in and out I'm getting 

```
Uncaught TypeError: Cannot read property '0' of undefined
```

I have traced this back to `wheeled` function in `zoom.js`. Specifically on [line 218](https://github.com/d3/d3-zoom/blob/cff5362ab4bea80c08ce28021b11d90eebd91509/src/zoom.js#L218) 

```javascript
g.zoom("mouse", constrain(translate(scale(t, k), mousePoint, mouseLocation), g.extent));
```

In my case `g.extent` is undefined.

**Comment by mbostock** [comment]
Please post an example that reproduces this issue. Otherwise I may not be
able to investigate.

**Comment by martinek** [comment]
I hit this while trying to figure out how to implement zooming so the code is somewhat dirty. There are also sample data I'm using.

https://gist.github.com/martinek/106a9159426c7c13f5a6a06224da20d9

**Comment by mbostock** [comment]
I’d say this is mostly a bug in your code, but maybe d3-zoom could handle this better. I wasn’t able to reproduce the error, but I did notice a bug bug investigating your code and I think it is the most likely cause.

The problem is that you’ve applied the zoom behavior twice: once to the G element and once to a child RECT element:

```js
const svg = d3.select(elem).append("svg")
    .attr("width", size.width)
    .attr("height", size.height)
    .append("g")
    .call(zoom);

// zoom panel
svg.append("rect")
    .attr('class', 'zoom-panel')
    .attr("width", size.width)
    .attr("height", size.height)
    .call(zoom);
```

So, when the wheel event is received, I think it’s possible that the zoom behavior could be confused: the zoom behavior isn’t able to handle multiple simultaneous wheel gestures, and the RECT and G element are treated as separate zoom gestures because you applied the zoom behavior to them separately.

I suspect this also has something to do with how the zoom behavior doesn’t consume wheel events that don’t trigger a change to zoom, thereby sometimes allowing both the RECT and G elements to start concurrent gestures, and then cause the race condition with the shared wheelTimer.

I think I can fix the error by making the wheelTimer state specific to the applied element rather than shared by the zoom behavior, but you’ll still want to fix your code so that the zoom behavior is only applied once.

Also… if you want your dates to parse across browsers you shouldn’t rely on Date.parse and instead use d3-time-format, e.g.,

```js
var parseTime = d3.timeParse("%Y-%m-%d %H:%M:%S %Z");;
```

**Comment by martinek** [comment]
Thanks for tips. That double zoom fixed some weird behavior I was getting.

**Comment by mbostock** [comment]
I pushed a fix in 1.0.2. If you have a chance to test it (with the double-zoom applied) and see if this fixes the TypeError, that would be helpful since I was never able to reproduce that error. You can load d3-zoom on top of the default D3 bundle:

```html
<script src="//d3js.org/d3.v4.js"></script>
<script src="//d3js.org/d3-zoom.v1.js"></script>
```

If you can’t, no worries. I’ll push a new D3 patch release 4.1.1 tomorrow most likely.
